### PR TITLE
Add support for organizations feature

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

This PR adds support for the (beta) Organizations and User Invitations features. Changes included:
- Supports specifying the `organization` and `invitation` query parameters when building the Authorization URL
- Supports verification of the `org_id` claim in `IdTokenVerifier`
- Adds documentation for using the Organizations and User Invitations features

### Testing

In addition to unit tests, the change was tested using an application and tenant with the Organizations feature enabled. I verified that with this change it adds the organization and invitation query parameters to the authorization URL to enable login/joining an organization.

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -315,6 +315,44 @@ try {
...
+- Manage their membership in a variety of ways, including user invitation.
+- Configure branded, federated login flows for each organization.
+- Implement role-based access control, such that users can have different roles when authenticating in the context of different organizations.
+- Build administration capabilities into your products, using Organizations APIs, so that those businesses can manage their own organizations.
+
+Note that Organizations is currently only available to customers on our Enterprise and Startup subscription plans.
+
+#### Log in to an organization
+
+Log in to an organization by using `withOrgganization()` when building the Authorization URL:
```
Comment (frederikprijck) [inline README.md:334]:
```suggestion
Log in to an organization by using `withOrganization()` when building the Authorization URL:
```

File: README.md [hunk 2]
```diff
@@ -315,6 +315,44 @@ try {
...
+
+```java
+AuthAPI auth = new AuthAPI("{YOUR_DOMAIN}", "{YOUR_CLIENT_ID}", "{YOUR_CLIENT_SECRET}");
+String url = auth.authorizeUrl("https://me.auth0.com/callback")
+    .withOrganization("{YOUR_ORGANIZATION_ID")
+    .build();
+```
+
+> When logging into an organization, it is important to ensure the `org_id` claim of the ID Token matches the expected organization value. The `IdTokenVerifier` can be configured to validate the `org_id` claim with the `withOrganization()` method on the `IdTokenVerifier.Builder`.
```
Comment (frederikprijck) [inline README.md:343]: Would it make sense to add an example on how to actually do this here?

File: src/main/java/com/auth0/client/auth/AuthorizeUrlBuilder.java [hunk 3]
```diff
@@ -105,6 +105,30 @@ public AuthorizeUrlBuilder withResponseType(String responseType) {
         return this;
     }
 
+    /**
+     * Sets the organization query string parameter value used to login to an organization.
+     *
+     * @param organization The ID of the organization to log the user into.
```
Comment (frederikprijck) [inline src/main/java/com/auth0/client/auth/AuthorizeUrlBuilder.java:111]:
```suggestion
* @param organization The ID of the organization to log the user in to.
```

File: src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java [hunk 4]
```diff
@@ -134,6 +136,15 @@ public void verify(String token, String nonce, Integer maxAuthenticationAge) thr
...
         }
 
+        // Org verification
+        if (this.organization != null) {
+            String orgClaim = decoded.getClaim("org_id").asString();
+            if (!this.organization.equals(orgClaim)) {
+                throw new IdTokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.organization, orgClaim));
+            }
+
+        }
```
Comment (frederikprijck) [inline src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java:146]: Should we have a comparable validation as we have with Audience?

- check if it exists and is a string
- Check if it matched ?

```suggestion
        if (this.organization != null) {
            String orgClaim = decoded.getClaim("org_id").asString();
            if (isEmpty(orgClaim)) {
               throw new IdTokenValidationException("Organization (org_id) claim must be a string present in the ID token");
            }
            if (!this.organization.equals(orgClaim)) {
                throw new IdTokenValidationException(String.format("Organization (org_id) claim mismatch in the ID token; expected \"%s\" but found \"%s\"", this.organization, orgClaim));
            }
        }
```

Edit: suggestion diff is a bit odd here ... I added 
```
if (isEmpty(orgClaim)) {
    throw new IdTokenValidationException("Organization (org_id) claim must be a string present in the ID token");
}
```

File: src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java [hunk 5]
```diff
@@ -447,6 +447,47 @@ public void succeedsWithValidTokenUsingDefaultClockAndHttpsDomain() {
...
+            .sign(Algorithm.HMAC256("secret"));
+
+        String jwt = JWT.decode(token).getToken();
+
+        configureVerifier(jwt)
+            .withOrganization("org_abc")
+            .build()
+            .verify(jwt);
+    }
+
```
Comment (frederikprijck) [inline src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java:490]: Should we also add the following two tests:

- fails when an organization is provided to the IdTokenValidator but no claim was found
- Doesn't fail when no organization is provided to IdTokenValidator but the token has an org_id claim

**Review by frederikprijck — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -315,6 +315,44 @@ try {
...
+### Accept user invitations
+
+Accept a user invitation by using `withInvitation()` when building the Authorization URL:
```
Comment (lbalmaceda) [inline README.md:346]: Is it worth mentioning the "organization" must also be passed along with the invitation?

File: README.md (hunk 2 above)
Comment (jimmyjames) [reply README.md:343]: Yes, I think that makes sense. I will add an example.

File: src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java (hunk 4 above)
Comment (jimmyjames) [reply src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java:146]: Yes, this makes sense. I'll push a new commit with these changes.

File: src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java (hunk 5 above)
Comment (jimmyjames) [reply src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java:490]: Yes, those are good tests to add, will push a new commit shortly.

File: README.md (hunk 6 above)
Comment (jimmyjames) [reply README.md:346]: It probably does make sense. @frederikprijck what do you think? Should we also update .NET to mention that "organization" must be set if specifying an invitation?

Comment (frederikprijck) [reply README.md:346]: Yes I can update it, I think its a good call

File: README.md (hunk 1 above)
Comment (jimmyjames) [reply README.md]: Pushed a new commit with this change

File: README.md (hunk 2 above)
Comment (jimmyjames) [reply README.md]: Example added in a new commit.

File: src/main/java/com/auth0/client/auth/AuthorizeUrlBuilder.java (hunk 3 above)
Comment (jimmyjames) [reply src/main/java/com/auth0/client/auth/AuthorizeUrlBuilder.java]: Change applied in a new commit.

File: src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java (hunk 4 above)
Comment (jimmyjames) [reply src/main/java/com/auth0/utils/tokens/IdTokenVerifier.java]: Added check for null/empty in new commit.

File: src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java (hunk 5 above)
Comment (jimmyjames) [reply src/test/java/com/auth0/utils/tokens/IdTokenVerifierTest.java:490]: Additional tests to cover those scenarios pushed, good catch!

File: README.md (hunk 6 above)
Comment (jimmyjames) [reply README.md:353]: I updated the JavaDoc to inform that organization must be specified when using `withInvitation()`.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**